File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        intervals.sort()
        res = 0
        prev = intervals[0][1]
        
        for start, end in intervals[1:]:
            if start >= prev:
                prev = end
            else:
                res += 1
                prev = min(end, prev)
        return res

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    const n = intervals.length
    intervals.sort((a, b) => a[1] - b[1])
    let result = 0
    let left = intervals[0]
    
    for (let i = 1; i < n; i++) {
        let right = intervals[i]
        if (right[0] < left[1]) {
            result += 1
        } else {
            left = right
        }
    }
    return result
};